app/controllers/posts_controller.rb

```diff
@@ -1,30 +1,40 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.where(user_id: params[:user_id])
+    @posts = Post.includes(:user).where(user_id: params[:user_id]).references(:posts)
     @user = User.find(params[:user_id])
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = Post.includes(:user).find(params[:id])
+    @comments = @post.comments.includes(:user)
   end
 
   def new
     @post = Post.new
+    render :new
   end
 
   def create
-    @post = Post.new(post_params)
-    @post.user_id = params[:user_id]
-    if @post.save
-      redirect_to user_posts_path
+    post = Post.new(params[:post].permit(:title, :text, :user_id))
+    post.user = current_user
+
+    if post.save
+      flash[:notice] = 'Post was created successfully.'
+      redirect_to user_posts_path(current_user, post)
     else
-      render :new
+      flash.now[:error] = 'Post was not created.'
+      render :show
     end
   end
 
-  private
-
-  def post_params
-    params.require(:post).permit(:text, :title, :user_id)
+  def destroy
+    post = Post.find(params[:id])
+    if post.destroy
+      flash[:notice] = 'Post was deleted successfully.'
+      redirect_to user_posts_path(post.user)
+    else
+      flash.now[:error] = 'Post was not deleted.'
+      render :show
+    end
```

Comment:
- [ ] I have created a post through the console to test if the authorization is working but when I tried to destroy the post I got the following error please make sure to fix it in order to test authorization 
<img width="1291" alt="Screen Shot 2022-10-04 at 17 56 32" src="https://user-images.githubusercontent.com/61790587/193880794-a454a13e-0b52-4e27-bde2-e1ba8a6ea3f6.png">